src/topology/separation.lean

```diff
@@ -667,12 +667,67 @@ end
   `x ≠ y` there exists disjoint open sets around `x` and `y`. This is
   the most widely used of the separation axioms. -/
 @[mk_iff] class t2_space (α : Type u) [topological_space α] : Prop :=
-(t2 : ∀x y, x ≠ y → ∃u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅)
+(t2 : ∀ x y, x ≠ y → ∃ u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅)
 
 lemma t2_separation [t2_space α] {x y : α} (h : x ≠ y) :
-  ∃u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅ :=
+  ∃ u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅ :=
 t2_space.t2 x y h
 
+lemma t2_separation_finset [t2_space α] : ∀ s : finset α,
+  ∃ f : α → set α, set.pairwise_disjoint ↑s f ∧ ∀ x : s, x.1 ∈ f x ∧ is_open (f x) :=
+begin
+  classical,
+  apply @finset.induction _ (λ s : finset α, ∃ f : α → set α, _) _,
+  { exact ⟨λ _, default, λ x, false.elim, λ x, x.2.elim⟩ },
+  { have t2_separation₂ : ∀ {x y}, x ≠ y → ∃ u : set α × set α, is_open u.1 ∧ is_open u.2 ∧
+      x ∈ u.1 ∧ y ∈ u.2 ∧ u.1 ∩ u.2 = ∅ := λ x y h, let ⟨u, v, H⟩ := t2_separation h in ⟨⟨u, v⟩, H⟩,
+    rintros t s ht ⟨f, hf, hf'⟩,
+    have hty : ∀ y : s, t ≠ y.1 := by { rintros y rfl, exact ht y.2 },
+    refine ⟨λ x, if ht : t = x then ⋂ y : s, (classical.some (t2_separation₂ (hty y))).1
+      else f x ∩ (classical.some (t2_separation₂ ht)).2, _, _⟩,
+    { rintros x hx₁ y hy₁ hxy a ⟨hx, hy⟩,
+      have hx₁ := finset.mem_insert.1 hx₁,
+      have hy₁ := finset.mem_insert.1 hy₁,
+      rcases eq_or_ne t x with rfl | hx₂;
+      rcases eq_or_ne t y with rfl | hy₂,
+      { exact hxy rfl },
+      { simp_rw [dif_pos rfl, mem_Inter] at hx,
+        simp_rw [dif_neg hy₂] at hy,
+        specialize hx ⟨y, hy₁.resolve_left hy₂.symm⟩,
+        have : a ∈ (classical.some (t2_separation₂ hy₂)).1 ∩ _ := ⟨hx, hy.2⟩,
+        rwa (classical.some_spec (t2_separation₂ hy₂)).2.2.2.2 at this },
+      { simp_rw [dif_neg hx₂] at hx,
+        simp_rw [dif_pos rfl, mem_Inter] at hy,
+        specialize hy ⟨x, hx₁.resolve_left hx₂.symm⟩,
+        have : a ∈ (classical.some (t2_separation₂ hx₂)).1 ∩ _ := ⟨hy, hx.2⟩,
+        rwa (classical.some_spec (t2_separation₂ hx₂)).2.2.2.2 at this },
+      { simp_rw [dif_neg hx₂] at hx,
+        simp_rw [dif_neg hy₂] at hy,
+        exact hf (hx₁.resolve_left hx₂.symm) (hy₁.resolve_left hy₂.symm) hxy ⟨hx.1, hy.1⟩ } },
+    { rintro ⟨x, hx⟩,
+      split_ifs with ht,
+      { have H := λ y, classical.some_spec (t2_separation₂ (hty y)),
+        refine ⟨mem_Inter.2 (λ y, _), is_open_Inter (λ y, (H y).1)⟩,
+        convert (H y).2.2.1,
+        exact ht.symm },
+      { have H := classical.some_spec (t2_separation₂ ht),
+        have hx := hf' ⟨x, (finset.mem_insert.1 hx).resolve_left (ne.symm ht)⟩,
+        exact ⟨⟨hx.1, H.2.2.2.1⟩, is_open.inter hx.2 H.2.1⟩ } } }
+end
+
+lemma t2_separation_finset' [t2_space α] {s : finset α} {x : α} (hx : x ∉ s) :
+  ∃ u : set α, is_open u ∧ disjoint u s :=
```

Comment:
This lemma doesn't seem particularly helpful - can't I always take `u` to be the empty set? Or do you want to say additionally that `x` is in `u`. In that case, I think this should work for t1 spaces: `s` must be closed since it's a finite union of singletons, so I can use its complement

Reply:
I missed the extra condition that `x ∈ u`. I'll fix that shortly.

Reply:
Oh, I see what you mean with T1 spaces. I guess this theorem isn't particularly useful after all, I'll remove it.

Reply:
If it's not there for T1 spaces I think it could be worth adding?

Reply:
Sure, I'll make this into a followup PR.

Reply:
Actually, `finset.is_closed` exists, so I don't think there's much use in this result.